demo/src/components/ToggleGroup.tsx

```diff
@@ -46,24 +85,37 @@ export function ToggleGroup<T extends string>({
               onClick={() => !item.disabled && onChange(item.value)}
               disabled={item.disabled}
               className={cn(
-                "w-full px-2 py-3 rounded-full transition-all duration-200 text-base font-inter",
-                variant === "vertical" ? "w-full text-center" : "",
+                "w-full p-2 rounded-full transition-all duration-200 text-base flex items-center justify-center gap-1.5",
+                variant === "vertical" && "w-full text-center",
                 value === item.value
-                  ? "bg-gray-50 text-[#6E938E] shadow-sm before:absolute before:inset-0 before:rounded-full before:border before:border-[#9DB5B2]/50"
+                  ? "bg-gray-50 text-[#004039] shadow-[0_1px_1px_1.5px_rgba(0,0,0,0.1)]"
                   : "text-[#6E938E] hover:text-[#004039]",
-                item.disabled
-                  ? "opacity-50 cursor-not-allowed hover:text-[#9DB5B2]"
-                  : ""
+                item.disabled &&
+                  "opacity-50 cursor-not-allowed hover:text-[#9DB5B2]",
+                shouldShowIcon(item.value) && "text-current"
               )}
             >
-              {item.label}
+              {shouldShowIcon(item.value) && (
+                <span
+                  className={cn(
+                    "inline-flex items-center justify-center w-[20px] h-[20px]",
+                    item.value === "default" ||
+                      item.value === "custom" ||
+                      item.value === "mainnet" ||
+                      item.value.includes("net")
+                      ? "text-current"
+                      : ""
+                  )}
+                >
+                  {getIcon(item.value)}
+                </span>
+              )}
+              <span>{item.label}</span>
             </button>
             {showTooltip && item.disabled && (
               <div className="absolute opacity-0 group-hover:opacity-100 transition-opacity duration-200 bottom-full left-1/2 -translate-x-1/2 mb-2 px-2 py-1 bg-gray-900 text-white text-xs rounded pointer-events-none whitespace-nowrap font-inter">
                 {item.value === "mainnet"
                   ? "Demo uses testnets only"
-                  : item.value === "custom"
-                  ? "Custom wallets coming soon"
```

Comment:
@guillaumegaluz did you intentionally remove this? I think for now we should keep until we fix custom wallets.

Reply:
not intentional. adding it back